code/modules/food&drinks/drinks/drinks.dm

```diff
@@ -288,3 +298,302 @@
+
+/obj/item/weapon/reagent_containers/food/drinks/dansoup/random
+	New()
+		..()
+		var/list/list = typesof(/obj/item/weapon/reagent_containers/food/drinks/dansoup) - list(/obj/item/weapon/reagent_containers/food/drinks/dansoup,/obj/item/weapon/reagent_containers/food/drinks/dansoup/random)
+		var/T = pick(list)
+		new T(loc)
+		spawn(0)
+			del src
+
+/obj/item/weapon/reagent_containers/food/drinks/dansoup/bbq
+	name = "Devil Dan's Quik-Noodles - Brimstone BBQ Flavor"
+	New()
+		..()
+		reagents.add_reagent("chicken_soup", 10)
+		reagents.add_reagent("soybeanoil", 1)
+		reagents.add_reagent("msg", 9)
+		reagents.add_reagent("sodiumchloride", 10)
+		reagents.add_reagent("nicotine", 8)
+		reagents.add_reagent("sulfur", 5)
+		reagents.add_reagent("beff", 5)
+//		reagents.add_reagent("ghostchili", 5)      uncomment once ghost chili juice is added
+
+/obj/item/weapon/reagent_containers/food/drinks/dansoup/macaroni
+	name = "Discount Dan's Quik-Noodles - Macaroni and Imitation Processed Cheese Product Flavor"
+	New()
+		..()
+		reagents.add_reagent("chicken_soup", 10)
+		reagents.add_reagent("soybeanoil", 9)
+		reagents.add_reagent("msg", 9)
+		reagents.add_reagent("sodiumchloride", 10)
+		reagents.add_reagent("nicotine", 8)
+		reagents.add_reagent("fake_cheese", 2)
+
+/obj/item/weapon/reagent_containers/food/drinks/dansoup/beef
+	name = "Comrade Dan's Quik-Noodles - Beef Perestroikanoff Flavor"
+	New()
+		..()
+		reagents.add_reagent("chicken_soup", 10)
+		reagents.add_reagent("soybeanoil", 1)
+		reagents.add_reagent("msg", 9)
+		reagents.add_reagent("sodiumchloride", 10)
+		reagents.add_reagent("nicotine", 8)
+		reagents.add_reagent("milk", 3)
+		reagents.add_reagent("enzyme", 3)
+		reagents.add_reagent("beff", 4)
+
+/obj/item/weapon/reagent_containers/food/drinks/dansoup/teriyaki
+	name = "Discount Deng's Quik-Noodles - Teriyaki TVP Flavor"
+	New()
+		..()
+		reagents.add_reagent("chicken_soup", 10)
+		reagents.add_reagent("hydrogenated_soybeanoil", 1)
+		reagents.add_reagent("msg", 14)
+		reagents.add_reagent("sodiumchloride", 10)
+		reagents.add_reagent("nicotine", 8)
+		reagents.add_reagent("synthflesh", 5)
+
+/obj/item/weapon/reagent_containers/food/drinks/dansoup/gamer
+	name = "Dan's Quik-Noodles - Gamer Grubs Flavor"
+	New()
+		..()
+		reagents.add_reagent("chicken_soup", 10)
+		reagents.add_reagent("soybeanoil", 1)
+		reagents.add_reagent("msg", 9)
+		reagents.add_reagent("sodiumchloride", 10)
+		reagents.add_reagent("nicotine", 8)
+		reagents.add_reagent("????", 10)
+		reagents.add_reagent("potassium", 5)
+
+/obj/item/weapon/reagent_containers/food/drinks/dansoup/mushroom
+	name = "Frycook Dan's Quik-Noodles - Mushroom-Swiss Burger-Bake Flavor"
+	New()
+		..()
+		reagents.add_reagent("chicken_soup", 10)
+		reagents.add_reagent("soybeanoil", 1)
+		reagents.add_reagent("msg", 9)
+		reagents.add_reagent("sodiumchloride", 10)
+		reagents.add_reagent("nicotine", 8)
+		reagents.add_reagent("beff", 2)
+		reagents.add_reagent("weird_cheese", 2)
```

Comment:
Apparently it's cheese made with special vomit, unlike normal cheese which is made with normal vomit